### PR TITLE
documentation improvements

### PR DESCRIPTION
**Pull request by cigrainger**
The aim of this PR is to improve documentation leading up to a hex release. I've:

- Significantly increased the moduledoc for `Explorer.DataFrame`
- Shrunk down the readme
- Changed the intro livebook to "ten minutes to explorer" and added it to the docs
- Added the numbat logo for ex_doc
- Made a few minor mods to `mix.exs` to prep for publishing

@kimjoaoun I'd love your input as well.

Once this is in I intend to publish 0.1.0 to Hex and set up a changelog.

### CODE REVIEW

**Review by polvalente — APPROVED** [review]

File: mix.exs [hunk 1]
```diff
@@ -8,11 +8,16 @@ defmodule Explorer.MixProject do
     [
       app: :explorer,
       name: "Explorer",
+      description: "Dataframes for Elixir.",
```
Comment (josevalim) [inline mix.exs:11]:
```suggestion
      description: "Series (one-dimensional) and dataframes (two-dimensional) fast data exploration for Elixir",
```

File: mix.exs [hunk 2]
```diff
@@ -25,16 +30,17 @@ defmodule Explorer.MixProject do
...
-      {:rustler_precompiled, "~> 0.2"},
+      {:rustler_precompiled, "~> 0.3"},
       {:table_rex, "~> 3.1.1"}
     ]
   end
 
   defp docs do
     [
       main: "Explorer",
+      logo: "numbat.png",
```
Comment (josevalim) [inline mix.exs:43]: We already have explorer.png, we can't reuse it?

Also make sure to `mix deps.update ex_doc` before. :D

Comment (cigrainger) [reply mix.exs:43]: We can! The numbat-only one looks a bit better in the top left IMO. I couldn't find where else the logo gets shown.

Comment (cigrainger) [reply mix.exs:44]: For comparison:

## Numbat only
image

## explorer.png
image

File: mix.exs (hunk 1 above)
Comment (cigrainger) [reply mix.exs]: What  about "Series (one-dimensional) and dataframes (two-dimensional) __for__ fast data exploration for Elixir"?

File: mix.exs (hunk 2 above)
Comment (josevalim) [reply mix.exs:44]: I see, in this case one with no padding whatsoever would likely be best! maybe call it explorer-exdoc.png or something :D

File: mix.exs (hunk 1 above)
Comment (josevalim) [reply mix.exs]: +1

**Pull request merged**